bigquery/google/cloud/bigquery/table.py

```diff
@@ -1099,3 +1145,101 @@ def to_dataframe(self):
         rows = [row.values() for row in iter(self)]
 
         return pandas.DataFrame(rows, columns=column_headers)
+
+
+class TimePartitioningType(object):
+    """The only type supported is DAY, which generates one partition per day.
+    """
+    DAY = 'DAY'
+
+
+class TimePartitioning(object):
+    """Configures time-based partitioning for a table.
+
+    Args:
+        partitioning_type (google.cloud.bigquery.table.TimePartitioningType):
+            The only type supported is DAY, which will generate one partition
+            per day.
+        field (str, optional):
+            If set, the table is partitioned by this field. If not set, the
+            table is partitioned by pseudo column '_PARTITIONTIME'. The field
+            must be a top-level TIMESTAMP or DATE field. Its mode must be
+            NULLABLE or REQUIRED.
+        expiration_ms(int, optional):
+            Number of milliseconds for which to keep the storage for a
+            partition.
+    """
+    def __init__(self, partitioning_type, field=None, expiration_ms=None):
+        self._properties = {}
+        self.partitioning_type = partitioning_type
+        if field is not None:
+            self.field = field
+        if expiration_ms is not None:
+            self.expiration_ms = str(expiration_ms)
```

Comment:
If you're using the property to set, you don't need to convert to `str` here.

Style-wise, I slightly prefer setting the whole `_properties` dictionary directly in the constructor, but it's fine to use the properties to set.